bundles/org.openhab.binding.vizio/src/main/java/org/openhab/binding/vizio/internal/handler/VizioHandler.java

```diff
@@ -0,0 +1,572 @@
+
+                    if (debounce && !polledInputState.getItems().isEmpty()
+                            && !currentInput.equals(polledInputState.getItems().get(0).getValue())) {
+                        currentInput = polledInputState.getItems().get(0).getValue();
+                        currentInputHash = polledInputState.getItems().get(0).getHashval();
+                        updateState(SOURCE, new StringType(currentInput));
+                    }
+                } catch (VizioException e) {
+                    logger.debug("Unable to retrieve Vizio TV current input. Exception: {}", e.getMessage(), e);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                }
+            }
+
+            if (powerOn && isLinked(ACTIVE_APP)) {
+                try {
+                    if (debounce) {
+                        CurrentApp polledApp = communicator.getCurrentApp();
+                        Optional<VizioApp> currentAppData = userConfigApps.stream()
+                                .filter(a -> a.getConfig().getAppId().equals(polledApp.getItem().getValue().getAppId())
+                                        && a.getConfig().getNameSpace()
+                                                .equals(polledApp.getItem().getValue().getNameSpace()))
+                                .findFirst();
+
+                        if (currentAppData.isPresent()) {
+                            if (!currentApp.equals(currentAppData.get().getName())) {
+                                currentApp = currentAppData.get().getName();
+                                updateState(ACTIVE_APP, new StringType(currentApp));
+                            }
+                        } else {
+                            currentApp = EMPTY;
+                            try {
+                                int appId = Integer.parseInt(polledApp.getItem().getValue().getAppId());
+                                updateState(ACTIVE_APP, new StringType(String.format(UNKNOWN_APP_STR, appId,
+                                        polledApp.getItem().getValue().getNameSpace())));
+                            } catch (NumberFormatException nfe) {
+                                // Non-numeric appId received, eg: hdmi1
+                                updateState(ACTIVE_APP, UnDefType.UNDEF);
+                            }
+
+                            logger.debug("Unknown app_id: {}, name_space: {}",
+                                    polledApp.getItem().getValue().getAppId(),
+                                    polledApp.getItem().getValue().getNameSpace());
+                        }
+                    }
+                } catch (VizioException e) {
+                    logger.debug("Unable to retrieve Vizio TV current running app. Exception: {}", e.getMessage(), e);
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
```

Comment:
Consider if the message above or even the exception message should be included in this status update to provide more immediate information to the user.